Pages/MicroPub/Endpoint.php

```diff
@@ -608,61 +611,65 @@ function deniedContent($title = '')
+                    case 'jpg':
+                    case 'jpeg':
+                        $mimetype = 'image/jpeg';
+                        break;
+                    case 'png':
+                        $mimetype = 'image/png';
+                        break;
+                    case 'gif':
+                        $mimetype = 'image/gif';
+                        break;
 
-                case 'mp3':
-                    $mimetype = 'audio/mpeg';
-                    break;
-                case 'oga':
-                case 'ogg':
-                    $mimetype = 'audio/ogg';
-                    break;
-                case 'wav':
-                    $mimetype = 'audio/x-wav';
-                    break;
-            }
+                    case 'mp4':
+                        $mimetype = 'video/mp4';
+                        break;
+                    case 'ogv':
+                        $mimetype = 'video/ogg';
+                        break;
 
-            $tmpname  = tempnam(sys_get_temp_dir(), 'indiepub_');
-            $fp       = fopen($url, 'rb');
-            if ($fp) {
-                $success = file_put_contents($tmpname, $fp);
-                fclose($fp);
-            } else {
-                $success = false;
-            }
-            if ($success) {
+                    case 'mp3':
+                        $mimetype = 'audio/mpeg';
+                        break;
+                    case 'oga':
+                    case 'ogg':
+                        $mimetype = 'audio/ogg';
+                        break;
+                    case 'wav':
+                        $mimetype = 'audio/x-wav';
+                        break;
+                }
 
-                // If we haven't got a valid mime, try using magicbyte detection
-                if (empty($mimetype)) {
-                    $finfo = finfo_open(FILEINFO_MIME_TYPE);
-                    $mimetype = finfo_file($finfo, $tmpname);
-                    finfo_close($finfo);
+                $tmpname  = tempnam(sys_get_temp_dir(), 'indiepub_');
+                $fp       = fopen($url, 'rb');
+                if ($fp) {
+                    $success = file_put_contents($tmpname, $fp);
+                    fclose($fp);
+                } else {
+                    $success = false;
                 }
+                if ($success) {
 
-                $_FILES[$type] = [
-                    'tmp_name' => $tmpname,
-                    'name'     => $pathinfo['basename'],
-                    'size'     => filesize($tmpname),
-                    'type'     => $mimetype,
-                ];
+                    // If we haven't got a valid mime, try using magicbyte detection
+                    if (empty($mimetype)) {
+                        $finfo = finfo_open(FILEINFO_MIME_TYPE);
+                        $mimetype = finfo_file($finfo, $tmpname);
+                        finfo_close($finfo);
+                    }
+
+                    $_FILES[$type][] = [
```

Comment:
Will this work with non photo types? Photos supports multiple uploads, but other stuff doesn't

Reply:
Non–photo micropub is untested. [micropub.rocks](https://micropub.rocks) does not provide uploads of any non–images